### PR TITLE
Update aws-sam-translator Dependency

### PR DESCRIPTION
**Pull request by ShreyaGangishetty**
*Issue #, if available:*

*Description of changes:*
Bump sam-translator version to 1.15.1
make pr fails as it cannot find 1.15.1 version

*Checklist:*

- [x] pytest tests/functional/commands/validate/lib/test_sam_template_validator.py  passes
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/functional/commands/validate/lib/models/basic_application.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Resources:
+  BasicApplication:
+    Type: 'AWS::Serverless::Application'
+    Properties:
+      Location:
+        ApplicationId: arn:aws:serverlessrepo:us-east-1:123456789012:applications/hello-world
+        SemanticVersion: 1.0.2
+
+  NormalApplication:
+    Type: 'AWS::Serverless::Application'
```
Comment (jfuss) [inline tests/functional/commands/validate/lib/models/basic_application.yaml]: Files with `AWS::Serverless::Application` types need to be removed. The `AWS::Serverless::Applicaiton` Resources require credentials in order for the Transform to work. For now, we have excluded these in the validation.

Comment (ShreyaGangishetty) [reply tests/functional/commands/validate/lib/models/basic_application.yaml]: Yes ... I have removed them manually in the next commit. I have to update the script to remove these files automatically from next time

**Comment by TheSriram** [comment]
Going to re-kick appveyor builds since #1465 is merged.

**Comment by TheSriram** [comment]
rebased on develop and pushed.

**Comment by TheSriram** [comment]
rebased on develop and fixed conflicts.

**Pull request merged**